### PR TITLE
SessionUser protection against nil pointer dereference

### PR DESCRIPTION
**Pull request by pboguslawski**
`SessionUser` should be protected against passing `sess` = `nil` to avoid

```
PANIC: runtime error: invalid memory address or nil pointer dereference
```

in

https://github.com/go-gitea/gitea/pull/18452/files#diff-a215b82aadeb8b4c4632fcf31215dd421f804eb1c0137ec6721b980136e4442aR69

after upgrade from gitea v1.16 to v1.17.

Related: https://github.com/go-gitea/gitea/pull/18452
Author-Change-Id: IB#1126459

### CODE REVIEW

File: services/auth/session.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2019 The Gitea Authors. All rights reserved.
+// Copyright 2022 The Gitea Authors. All rights reserved.
```
Comment (techknowlogick) [inline services/auth/session.go]: thanks for this PR, no need to update the date of copyright, we keep it as the date of when the file was first created, and then we rely on git blame for when the file is updated.

Comment (pboguslawski) [reply services/auth/session.go]: Fixed.

**Review by Gusted — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Comment by lunny** [comment]
Please send backport

**Comment by pboguslawski** [comment]
PR with backport: #21358

**Pull request merged**